sdk/storage/azure-storage-blob/azure/storage/blob/_list_blobs_helper.py

```diff
@@ -7,14 +7,70 @@
 
 from azure.core.paging import PageIterator, ItemPaged
 from azure.core.exceptions import HttpResponseError
+
 from ._deserialize import get_blob_properties_from_generated_code, parse_tags
-from ._generated.models import BlobItemInternal, BlobPrefix as GenBlobPrefix, FilterBlobItem
+from ._generated.models import FilterBlobItem
 from ._models import BlobProperties, FilteredBlob
 from ._shared.models import DictMixin
+from ._shared.xml_deserialization import unpack_xml_content
 from ._shared.response_handlers import return_context_and_deserialized, process_storage_error
 
 
-class BlobPropertiesPaged(PageIterator):
+def deserialize_list_result(pipeline_response, *_):
+    payload = unpack_xml_content(pipeline_response.http_response)
```

Comment:
I _believe_ this line here is replacing the ContentDecodePolicy that I removed from the pipeline. So this would already be unpacked if we put that policy back in.